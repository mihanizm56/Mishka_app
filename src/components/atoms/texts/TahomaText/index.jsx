// @flow
import React, { Node, createRef } from "react";
import "./TahomaText.css";
import { getClass } from "../../../utils";

type TahomaTextProps = {
    text: string,
    bold?: boolean,
    classname?: string,
    hovered?: boolean,
    inlineStyles?: Object,
    ligth?: boolean,
    title?: string,
    needRef?: boolean,
    handleClick?: Function,
    withRefClick?: Function
};

export const TahomaText = ({
    text,
    bold,
    inlineStyles,
    classname,
    hovered,
    ligth,
    title,
    needRef,
    handleClick,
    withRefClick
}: TahomaTextProps): Node => {
    const OpenSansRef = needRef ? createRef() : null;

    return (
        <p
            style={{ ...inlineStyles }}
            title={title}
            className={getClass({ initialClass: 'tahoma-text', bold: bold, hovered: hovered, ligth: ligth, additionalClass: classname})}
            ref={needRef && OpenSansRef}
            onClick={needRef && withRefClick ? () => withRefClick(OpenSansRef) : handleClick}
        >
            {text}
        </p>
    )
}


TahomaText.defaultProps = {
    text: 'default Tahoma text'
}
