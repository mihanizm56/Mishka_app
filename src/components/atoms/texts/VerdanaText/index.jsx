//
import React, { Node, createRef } from "react";
import "./VerdanaText.css";
import { getClass } from "../../../../utils";

type VerdanaTextProps = {
	text: string,
	bold?: boolean,
	classname?: string,
	hovered?: boolean,
	inlineStyles?: Object,
	ligth?: boolean,
	title?: string,
	needRef?: boolean,
	handleClick?: Function,
	withRefClick?: Function,
};

export const VerdanaText = (props: VerdanaTextProps): Node => {
	const { text, bold, inlineStyles, classname, hovered, ligth, title, needRef, handleClick, withRefClick } = props;
	const OpenSansRef = needRef ? createRef() : null;

	return (
		<p
			style={{ ...inlineStyles }}
			title={title}
			className={getClass({
				initialClass: "verdana-text",
				bold: bold,
				hovered: hovered,
				ligth: ligth,
				additionalClass: classname,
			})}
			ref={needRef && OpenSansRef}
			onClick={needRef && withRefClick ? () => withRefClick(OpenSansRef) : handleClick}
		>
			{text}
		</p>
	);
};

VerdanaText.defaultProps = {
	text: "default Verdana text",
};
