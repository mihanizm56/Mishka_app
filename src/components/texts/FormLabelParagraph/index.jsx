// @flow
import React, { Node } from "react";
import { Paragraph } from "../";
import "./FormLabelParagraph.css";

type FormLabelParagraphProps = {
	text: string,
};

export const FormLabelParagraph = ({ text }: FormLabelParagraphProps): Node => (
	<Paragraph text={text} className="title-form-label-paragraph" />
);

FormLabelParagraph.defaultProps = {
	text: "default form title label",
};
