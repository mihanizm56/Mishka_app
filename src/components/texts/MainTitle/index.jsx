import React from "react";
import "./MainTitle.css";

export const MainTitle = ({ text }) => <h1 className="main-title">{text}</h1>;
