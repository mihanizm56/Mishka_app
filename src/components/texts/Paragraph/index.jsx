// @flow
import React, { Node, createRef } from "react";

type ParagraphProps = {
	text: string,
	className: string,
};

export const Paragraph = ({ text, className, customFontSize, title, handleClick, needRef }: ParagraphProps): Node => {
	
	const ParagraphRef = needRef ? createRef() : null;

	return (
		<p className={className} ref={needRef && ParagraphRef} title={title} style={customFontSize ? { fontSize: customFontSize } : null} onClick={()=>handleClick(ParagraphRef)}>
			{text}
		</p>
	)
}

Paragraph.defaultProps = {
	text: 'default', 
	handleClick: (value)=>value
}