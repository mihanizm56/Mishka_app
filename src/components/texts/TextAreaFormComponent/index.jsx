// @flow
import React, { Node } from "react";
import { getClass } from "../../../helpers";
import "./TextAreaFormComponent.css";

type TextAreaProps = {
	placeholder: string,
	width: string,
	height: string,
};

export const TextAreaFormComponent = ({ placeholder, width, height, isError }: TextAreaProps): Node => (
	<textarea className={getClass({ initialClass: "textarea", error: isError })} placeholder={placeholder} />
);

TextAreaFormComponent.defaultProps = {
	placeholder: "default textarea placeholder",
};
