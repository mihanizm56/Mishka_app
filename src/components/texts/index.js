export * from "./MainTitle";
export * from "./NameForReview";
export * from "./Paragraph";
export * from "./ParagraphForCards";
export * from "./TextAreaFormComponent";
export * from "./BoldParagraph";
export * from "./FooterTextLogo";
export * from "./TextForHeaderBasketIcon";
export * from "./normal";
