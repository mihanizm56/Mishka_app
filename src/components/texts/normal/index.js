export * from './OpenSansText';
export * from './TahomaText';
export * from './VerdanaText';