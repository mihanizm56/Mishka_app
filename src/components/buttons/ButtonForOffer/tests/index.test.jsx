import React from 'react';
import {shallow} from 'enzyme';
import {ButtonForOffer} from '../';
import {TextForButton} from '../../../texts';

describe('ButtonForOffer test', () => {
    const testString = 'test string';
    const mockClick = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
    });

    describe('test render', () => {
        it('render div wrapper', () => {
            const component = shallow(<ButtonForOffer text={testString} callback={mockClick}/>);
            expect(component.find('.button-offer-wrapper').length).toEqual(1);
        });
        it('render text component', () => {
            const component = shallow(<ButtonForOffer text={testString} callback={mockClick}/>);
            expect(component.find(TextForButton).length).toEqual(1);
        });
    })

    describe('test props', () => {
        it('check text prop', () => {
            const component = shallow(<ButtonForOffer text={testString} />);
            expect(component.find(TextForButton).props().text).toEqual(testString);
        });
        it('check callback prop', () => {
            const component = shallow(<ButtonForOffer callback={mockClick} />);
            expect(component.find('.button-offer-wrapper').props().onClick).toEqual(mockClick);
        });
    })

    describe('action of component', () => {
        it('check callback on click', () => {
            const component = shallow(<ButtonForOffer callback={mockClick}/>);
            component.find('.button-offer-wrapper').simulate('click');
            expect(mockClick).toHaveBeenCalled();
        });
    })
})
    