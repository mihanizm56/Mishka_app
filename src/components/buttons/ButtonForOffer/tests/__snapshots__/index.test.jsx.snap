// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonForOffer test test render render snap without any props 1`] = `
<div
  className="button-offer-wrapper"
  onClick={[Function]}
>
  <TextForButton
    text="СДЕЛАТЬ ЗАКАЗ"
  />
</div>
`;
